src/en/developer-event-cycle.md

```diff
@@ -1,14 +1,82 @@
-Title: Event Cycle  
+Title: Event Cycle
 
 # Event Cycle
 
+When charming with the reactive framework, it’s possible to use
+developer-created states and boolean logic to run code when the states represent
+something meaningful to your deployment, such as when a database is connected,
+and when it is available, or when the primary workload becomes available for
+use. These are additional “events” that are abstracted from the hook event cycle
+outlined later in this page.
+
+## Handling reactive states
+
+Many states can be performed during a single hook execution, as these abstracted
```

Comment:
I agree with you. The page would benefit greatly from a full edit, particularly with assistance from a subject matter expert.